api-controllers/FabrikBaseController.js

```diff
@@ -49,24 +68,38 @@ class FabrikBaseController extends BaseController {
     };
   }
 
-  _lockResource(req, operationType) {
+  _lockResource(req, operationType, serviceFlowId) {
     const plan_id = req.body.plan_id || req.query.plan_id;
     const plan = catalog.getPlan(plan_id);
     return Promise.try(() => {
       if (plan.manager.name === CONST.INSTANCE_TYPE.DIRECTOR) {
         // Acquire lock for this instance
         return lockManager.lock(req.params.instance_id, {
-          lockedResourceDetails: {
-            resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
-            resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
-            resourceId: req.params.instance_id,
-            operation: operationType
-          }
+          lockedResourceDetails: this._getLockResourceDetails(req, operationType, serviceFlowId)
         }, plan);
       }
     });
   }
 
+  _getLockResourceDetails(req, operationType, serviceFlowId) {
+    if (_.includes(CONST.OPERATION_TYPE.LIFECYCLE, operationType)) {
+      if (serviceFlowId !== undefined) {
+        return {
+          resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.SERVICE_FLOW,
+          resourceType: CONST.APISERVER.RESOURCE_TYPES.SERIAL_SERVICE_FLOW,
+          resourceId: serviceFlowId,
+          operation: operationType
+        };
+      }
+      return {
+        resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.DEPLOYMENT,
+        resourceType: CONST.APISERVER.RESOURCE_TYPES.DIRECTOR,
+        resourceId: req.params.instance_id,
+        operation: operationType
+      };
+    }
+  }
+
   _unlockIfReqfailed(operationType, processedRequest, lockId, req, res, next, err) {
     const plan_id = req.body.plan_id || req.query.plan_id;
```

Comment:
existing code not modified in this PR. If one were to fix it truly, since plan_id is anyways required in the flow below, it must throw an BadRequest Error when its null.